docs/index.bs

```diff
@@ -2645,16 +2652,15 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
                       Note: The control does not break the loop in this step to continue with all the imported scripts to populate the cache.
           1. Asynchronously complete these steps with |response|.
 
-          If the algorithm asynchronously completes with null, then:
+          When the algorithm asynchronously completes, continue the rest of these steps, with |script| being the asynchronous completion value.
 
-              1. Invoke [=Reject Job Promise=] with |job| and `TypeError`.
-
-                  Note: This will do nothing if [=Reject Job Promise=] was previously invoked with "{{SecurityError}}" {{DOMException}}.
+      1. If |script| is null or [=Is Async Module=] with |script|'s [=script/record=], |script|'s [=script/base URL=], and « » is true, then:
```

Comment:
A random idea: How about checking whether `|evaluationPromise|.\[[PromiseState]] is "pending"` instead of `[=Is Async Module=] ... is true`?

This makes the logic a little simpler (using only `|evaluationPromise|.\[[PromiseState]]`, removing module graph traversal and module inspection), but this is not equivalent from the current condition (e.g. for modules with `await undefined`).

Reply:
A goal of the design here was to not be "racy" in the sense that, both asynchronous behavior and errors are based on the module syntax, not whether it has already run or not, or whether it dynamically reached an `await`. This would make behavior more stable.

Reply:
Thanks, it makes sense > based on syntax.